.github/workflows/linux.yaml

```diff
@@ -0,0 +1,28 @@
+name: linux
+on: [pull_request]
+
+jobs:
+  build:
+    name: build and test
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: dependencies
+        shell: bash
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y -qq make cmake gcc g++
+      - name: build
+        shell: bash
+        run: |
+          export LD_LIBRARY_PATH=.:$(cat /etc/ld.so.conf.d/* | grep -vF "#" | tr "\\n" ":" | sed -e "s/:$//g")
+          cmake . -DCMAKE_BUILD_TYPE=RelWithDebInfo -DENABLE_SANITIZERS=On && make all -j$(nproc)
+      - name: test
+        shell: bash
+        run: |
+          export LD_LIBRARY_PATH=.:$(cat /etc/ld.so.conf.d/* | grep -vF "#" | tr "\\n" ":" | sed -e "s/:$//g")
+          #TODO: run statsd to collect these test metrics
+          ./testStatsdClient
+          #TODO: validate test metrics received
```

Comment:
i think we should flesh this bit out a little and actually run statsd and check that it got all the test metrics. im new to statsd but if we could figure something to just dump them to file it would be pretty straight forward

Reply:
The current test suite is very shallow indeed. I won't have the time to setup better tests now but this is definitely something worth working on. 
If you already have good ideas, feel free to suggest in another PR or raise an issue.

Reply:
I'd be happy to contribute them as well in the next PR!

Reply:
i see that statsd has a "console" backend which should write to file if we run it correctly. i think this would allow us to do a proper "integration" test in CI. we should add unit tests too of course

Reply:
We can discuss this further in the next PR but running something like this with docker-compose and calling the graphite API should be simple enough:

```
gostatsd:
  image: atlassianlabs/gostatsd:29.0.0
  command: gostatsd --config-path=/etc/gostatsd/config.toml
  links:
    - graphite
  ports:
    - "8125:8125/udp"
    - "8126:8126"
    - "8181:8181"
  volumes:
    - ${PWD}:/etc/gostatsd:ro

graphite:
  image: sitespeedio/graphite
  ports:
    - "8080:80"
    - "8000:8000"
    - "2003:2003"
    - "2004:2004"
    - "7002:7002"
```

Reply:
interesting! i was able to loaded up statsd directly on ubuntu and configure the console backend and see that it spit out the various stats that the test program sent, in javascript/json notation. i didnt update the CI to check it though. would you prefer graphite? and if so, how does one retrieve the values then to make sure they are what they should be?

Reply:
I would have used [Graphite's HTTP API](https://graphite-api.readthedocs.io/en/latest/api.html#metrics-find) to check that the metrics were received but I'm fine with using only statsd.
Having this running in the CI is probably worth another PR though.